src/main/java/uk/gov/di/ipv/cri/common/library/util/SignedJWTFactory.java

```diff
@@ -8,16 +8,37 @@
 import com.nimbusds.jose.util.Base64URL;
 import com.nimbusds.jwt.JWTClaimsSet;
 import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.codec.binary.Hex;
 
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.text.ParseException;
 
 public class SignedJWTFactory {
+
+    private static final String KID_PREFIX = "did:web:";
     private final JWSSigner kmsSigner;
 
     public SignedJWTFactory(JWSSigner kmsSigner) {
         this.kmsSigner = kmsSigner;
     }
 
+    // Updated method for passing issuer used in building kid
+    public SignedJWT createSignedJwt(
```

Comment:
I dont think we actually need `keyIDInHeader`
As were using a new method with issuer and keyid the fact were going to add a kid should be implied